libindy/src/commands/crypto.rs

```diff
@@ -397,15 +397,73 @@ impl CryptoCommandExecutor {
         // Use AEAD to encrypt `message` with "protected" data as "associated data"
         let (ciphertext, iv, tag) =
             self.crypto_service
-                .encrypt_plaintext(message, &base64_protected, &cek);
+                .encrypt_plaintext(message, &json_str, &cek);
 
         self._format_pack_message(&base64_protected, &ciphertext, &iv, &tag)
     }
 
+    pub fn unpack_msg(&self, jwe_json: Vec<u8>, wallet_handle: WalletHandle) -> IndyResult<Vec<u8>> {
```

Comment:
@lovesh please restore initial location of this function against others in the file. I would like to have clear diff for discussion in progress. And when we will be ready to merge - we can change the order